src/Cake.Common.Tests/Unit/Build/TFBuild/TFBuildCommandTests.cs

```diff
@@ -463,22 +465,115 @@ public void Should_Publish_Test_Results()
                     PublishRunAttachments = true,
                     TestRunner = TFTestRunnerType.XUnit,
                     TestRunTitle = "Cake Test Run 1 [master]",
-                    TestResultsFiles = new string[]
+                    TestResultsFiles = new FilePath[]
                      {
-                         FilePath.FromString("./artifacts/resultsXUnit.trx").ToString(),
-                         FilePath.FromString("./artifacts/resultsJs.trx").ToString()
+                         new FilePath("./artifacts/resultsXUnit.trx"),
+                         new FilePath("./artifacts/resultsJs.trx")
                      }
                 };
 
                 // When
                 service.Commands.PublishTestResults(data);
 
                 // Then
-                const string expected = @"##vso[results.publish type=XUnit;mergeResults=true;platform=x86;config=Debug;runTitle='Cake Test Run 1 [master]';publishRunAttachments=true;resultFiles=C:\build\CAKE-CAKE-JOB1\artifacts\resultsXUnit.trx,C:\build\CAKE-CAKE-JOB1\artifacts\resultsJs.trx;]";
+                const string expected = @"##vso[results.publish type=XUnit;mergeResults=true;platform=x86;config=Debug;runTitle='Cake Test Run 1 [master]';publishRunAttachments=true;resultFiles=C:/build/CAKE-CAKE-JOB1/artifacts/resultsXUnit.trx,C:/build/CAKE-CAKE-JOB1/artifacts/resultsJs.trx;]";
```

Comment:
There need to be tests that run on Windows that fail if the slash is `/` and tests that run on Linux that fail if the slash is `\`. Or to simplify, we need a test that runs on all platforms that asserts that only `System.IO.Path.DirectorySeparatorChar` appears in the output.

Reply:
@jnm2 Your saying that the two file paths are mutually exclusive in this TFS regardless of what the operating system can handle?  I understand why the original author added that, but it clearly never worked for anything other than windows.

It's been a few weeks, but I remember testing this on a AzureDevOps Mac Agent, and it worked.

I agree that the default `FilePath` or `DirectoryPath` should resolve down to the native platform, and that being the best solution.  So should I wait for your PR to get merged and then use that concept?

Reply:
> Your saying that the two file paths are mutually exclusive in this TFS regardless of what the operating system can handle? I understand why the original author added that, but it clearly never worked for anything other than windows.

Yes, that is what I recall from having implemented this in my own code and it's the only reason the original author added the Replace. You're right that it clearly was Windows-only, but at the time, we weren't aware that if Visual Studio Team Services ever offered macOS or Linux builds.

> So should I wait for your PR to get merged and then use that concept?

It's up to you. I think I'll submit a PR when I get a chance, but you always have the option of doing `.Replace(filePath.Separator, System.IO.Path.DirectorySeparatorChar)` in the meantime just to get this working.

Reply:
@jnm2 So I made these changes you suggested.  I am unsure why it doesn't work the way I expect when I set the testing fixture to return `PlatformFamily.OSX` I still get the output with `\` as opposed to `/` as I would expect.  Do I need a `OSXFact` so it will only execute on a non windows platform?

Reply:
@RLittlesII Oh my, does `System.IO.Path.DirectorySeparatorChar` produce `\` on macOS? I didn't know.

Reply:
I am not sure if the test is the problem.  More so when I use `PlatformFamily.OSX` I get `\` instead of `/` and I wanted to be sure that was expected behavior based on what you know.

So I need to write a test that is a bit more dynamic that the current static expected result.  I believe I understand, ping you when I've updated.  Thanks!

Reply:
@jnm2 Does the reverted code still cause a problem based on your original comments?

Reply:
Well, you can't hardcode an expected `/` if the test runs on Windows. Like I pointed out in my original comment, `/` causes TFS on Windows to fail.

https://github.com/cake-build/cake/pull/2339/files#diff-37bb9b9e900501c59faf81cfa4cfda62R539

Reply:
And https://github.com/cake-build/cake/pull/2339/files#diff-37bb9b9e900501c59faf81cfa4cfda62R509, too.

I'd set the expected string to `.Replace(filePath.Separator, System.IO.Path.DirectorySeparatorChar)` to avoid having to write separate Windows facts and anything-but-Windows facts.

Reply:
I now only have a single windows fact for the Absolute Path.  Primarily to keep from having to do too much parsing magic in the test.